### PR TITLE
prometheus: add WithNamespace option to prefix metrics

### LINKED ISSUE #3437: Add a `namespace` option to the Prometheus Exporter
### Problem Statement

As an user of the Prometheus Exporter, I would like to be able to set a namespace `prefix` for my exported metrics. 

### Proposed Solution

A new `prometheus.WithNamespace()` option that would add the given namespace to all metrics except meta metrics, i.e., `target_info` metrics shouldn't be prefixed since they have some special behavior.

#### Alternatives

One could use a prometheus wrapped registerer that is passed for the Prometheus Exporter, e.g, `prometheus.WrapRegistererWithPrefix(namespace, registry)`, but that adds the namespace prefix to the `target_info` metrics as well.

#### Prior Art

This is a common option existing on various prometheus exporter including Otel Collector's Contrib [Prometheus Remote Write Exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/prometheusremotewriteexporter#getting-started) and [Prometheus Exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/prometheusexporter#getting-started), OpenCensus and other prometheus libraries as well.

### PR DESCRIPTION
**Pull request by paivagustavo**
Adds a `WithNamespace` option to the prometheus exporter.

Currently, the lack of such option blocks the otel collector: https://github.com/open-telemetry/opentelemetry-collector/issues/7454

To finalize the collector migration to otel-go and start enabling it by default, both instrumentations must be compatible. 

Collector's OpenCensus instrumentation [prefixes metrics with](
https://github.com/open-telemetry/opentelemetry-collector/blob/6b2198eb709a5d498d6b1347c437f19c8adbb9a8/service/telemetry.go#L181) the `otelcol_` prefix.
To match that behavior in the otel-go instrumentation, we [tried to wrap](https://github.com/open-telemetry/opentelemetry-collector/blob/6b2198eb709a5d498d6b1347c437f19c8adbb9a8/service/telemetry.go#L214) a `prometheus.Registry` with the `otelcol_` prefix. 

But since Otel Go resource attributes are exported as part of the `target_info` metric, which is also prefixed and becomes `otelcol_target_info`, for example:
```
# HELP otelcol_otel_scope_info Instrumentation Scope metadata
# TYPE otelcol_otel_scope_info gauge
otelcol_otel_scope_info{otel_scope_name="testmeter",otel_scope_version="v0.1.0"} 1
# HELP otelcol_target_info Target metadata
# TYPE otelcol_target_info gauge
otelcol_target_info{service_name="prometheus_test",telemetry_sdk_language="go",telemetry_sdk_name="opentelemetry",telemetry_sdk_version="latest"} 1
```

This make it impossible for the OpenTelemetry Collector's Prometheus Receiver to properly parse the `target_info` and `otel_scope_info` into the appropriated resource attributes and scope attributes. For that reason, the Otel Go instrumentation is not compatible with the existing OpenCensus instrumentation.

Adding such a common option in the prometheus instrumentation would unblock the Collector's migration to Otel Go and also other external users such as: https://github.com/open-telemetry/opentelemetry-go/issues/3405#issuecomment-1335518018

For reference, this option is also available on collector's [prometheus exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/prometheusexporter/config.go#L31-L32)

closes #3437

### CODE REVIEW

File: exporters/prometheus/exporter.go [hunk 1]
```diff
@@ -316,9 +318,12 @@ var unitSuffixes = map[string]string{
 	"ms": "_milliseconds",
 }
 
-// getName returns the sanitized name, including unit suffix.
+// getName returns the sanitized name, prefixed with the namespace and suffixed with unit.
 func (c *collector) getName(m metricdata.Metrics) string {
 	name := sanitizeName(m.Name)
+	if c.namespace != "" {
+		name = c.namespace + name
+	}
```
Comment (MadVikingGod) [inline exporters/prometheus/exporter.go:326]: Should we do this before the name is sanitized?

**Review by MadVikingGod — APPROVED** [review]

File: exporters/prometheus/exporter.go (hunk 1 above)
Comment (paivagustavo) [reply exporters/prometheus/exporter.go:326]: That is a good question but I've opted to sanitize the namespace once when in the `WithNamespace()` method, so `c.namespace` here is already sanitized. This avoid the need to sanitize the namespace multiple times.

**Review by jmacd — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 - The `go.opentelemetry.io/otel/metric/embedded` package. (#3916)
 - The `Version` function to `go.opentelemetry.io/otel/sdk` to return the SDK version. (#3949)
+- Add a `WithNamespace` option to `go.opentelemetry.io/otel/exporters/prometheus` to allow users to prefix metrics with a namespace (#TBD).
```
Comment (pellared) [inline CHANGELOG.md:15]:
```suggestion
- Add a `WithNamespace` option to `go.opentelemetry.io/otel/exporters/prometheus` to allow users to prefix metrics with a namespace. (#3970)
```

**Review by pellared — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (paivagustavo) [reply CHANGELOG.md]: Thank you! :)

**Review by Aneurysm9 — APPROVED** [review]

**Pull request merged**